website/source/docs/guides/consul-splitting.md

```diff
@@ -0,0 +1,461 @@
+
+Once everything is up and running, you can view the health of the registered
+services by checking the Consul UI at `http://localhost:8500`. All services
+should be passing their health checks.
+
+![List of services in the Consul UI including Consul, and the web and API services with their proxies](/static/img/consul-splitting-services.png)
+
+Curl the Web endpoint to make sure that the whole application is running. The
+Web service will get a response from version 1 of the API service.
+
+```hcl
+$ curl localhost:9090
+Hello World
+###Upstream Data: localhost:9091###
+  Service V1%
+```
+
+Initially, you will want to deploy version 2 of the API service to production
+without sending any traffic to it, to make sure that it performs well in a new
+environment. Prevent traffic from flowing to version 2 when you register it, you
+will preemptively set up a  traffic split to send 100% of your traffic to
+version 1 of the API service, and 0% to the not-yet-deployed version 2.
+Splitting the traffic makes use of the new Layer 7 features built into Consul
+Service Mesh.
+
+To deploy version 2 of your API service, you will:
+
+1. Set up a traffic split to make sure v2 doesn’t receive any traffic at first.
+1. Start an instance of the v2 API service in your production environment.
+1. Register v2 so that Consul can send traffic to it.
+1. Slowly shift traffic to v2 and a way from v1 until the new version is
+handling all the traffic.
+
+## Configure Traffic Splitting
+
+Traffic Splitting uses configuration entries (introduced in Consul 1.5 and 1.6)
+to centrally configure the services and Envoy proxies. There are three
+configuration entries you need to create to enable traffic splitting:
+
+- Service Defaults for the API service to set the protocol to HTTP.
+- Service Splitter which defines the traffic split between the service subsets.
+- Service Resolver which defines which service instances are version 1 and 2.
+
+### Configuring Service Defaults
+
+Traffic splitting requires that the upstream application uses HTTP, because
+splitting happens on layer 7 (on a request by request basis). You will tell
+Consul that your upstream service uses HTTP by setting the protocol in a
+“service defaults” configuration entry for the API service. This configuration
+is already in your demo environment at `l7_config/api_service_defaults.json`. It
+contains the following.
+
+```json
+{
+  "kind": "service-defaults",
+  "name": "api",
+  "protocol": "http"
+}
+```
+
+The `kind` field denotes the type of configuration entry which you are defining;
+for this example, `service-defaults`. The `name` field defines which service the
+service-defaults configuration entry applies to. (The value of this field must
+match the name of a service registered in Consul, in this example, `api`.) The
+`protocol` is `http`.
+
+To apply the configuration, you can either use the Consul CLI or the API. In
+this example we’ll use the configuration entry endpoint of the HTTP API, which
```

Comment:
Oh interesting, I didn't know that we had a preference. 

Usually, I default to the HTTP API because I assume that when users have big datacenters they might not want to log into the right node to register a new service or check, etc. 

Not sure what a real-life workflow looks like, but we can definitely default to the CLI from now on 😄

Reply:
Yeah I think HTTP does make some sense for that reason but I think our CLI usually offers a better experience so we should maybe try to steer towards using that where possible. In other words, we have more control over the UX with the CLI (c.f. creating intentions with `consul intentions create -allow foo bar` vs trying to do that with JSON and curl.)

It's def possible to use consul CLI from a bastion host or similar but we perhaps don't spell that out anywhere. Maybe we should? By default it connects to localhost because it expects every machine to be running an agent, but it never needs to be run from the actual server hosts and if it's on a host with no agent you can point it to another host via ENV of args easily enough.

What do you think?

Reply:
I think having a guide about connecting the CLI would be great. Putting it on our to-do list now so it doesn't get lost. 

The one other advantage that I just thought of with giving the HTTP API example is that I assume people are going to want to automate most stuff. As an example, Nic's original blog was really canary focused, and my understanding is that sometimes people will automate those based on pre-defined thresholds. I assume organizations that are actually that sophisticated are few and far between though.

In any case, I'll replace the HTTP examples here with CLI instead 😄 Nic updated the application a little bit so this will give me an excuse to retest with the updates.